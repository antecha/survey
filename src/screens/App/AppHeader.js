import React from 'react';
import {Layout, Icon} from 'antd';


const AppHeader = ({collapsed,onIconClick}) => <Layout.Header style={{ background: '#fff', padding: 0 }}>
<Icon
  className="trigger"
  type={collapsed ? 'menu-unfold' : 'menu-fold'}
  onClick={onIconClick}
/>
</Layout.Header>

export default AppHeader;