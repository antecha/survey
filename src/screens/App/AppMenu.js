import React from 'react';
import {  Layout, Menu, Icon } from 'antd';
import { Link } from 'react-router-dom'

const {Sider} = Layout;

const AppMenu = ({collapsed})=> <Sider trigger={null} collapsible collapsed={collapsed}>
<div className="logo" />
<Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
  <Menu.Item key="1">
    <Icon type="user" />
    <span>home</span>
    <Link to="/" />
  </Menu.Item>
  <Menu.Item key="2">
    <Icon type="video-camera" />
    <span>nav 2</span>
    <Link to="/nav2" />
  </Menu.Item>
  <Menu.Item key="3">
    <Icon type="upload" />
    <span>nav 3</span>
    <Link to="/nav3" />
  </Menu.Item>
</Menu>
</Sider>

export default AppMenu;