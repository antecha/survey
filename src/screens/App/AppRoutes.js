import React from 'react';
import { Route } from 'react-router-dom';
import { Layout } from 'antd';

const { Content } = Layout;

const AppContent = ({ text }) => (
  <Content
    style={{
      margin: '24px 16px',
      padding: 24,
      background: '#fff',
      minHeight: 280,
    }}
  >
    {text}
  </Content>
);

const AppRoutes = () => (
  <React.Fragment>
    <Route exact path="/">
      <AppContent text="Survey List" />
    </Route>

    <Route path="/create-survey">
      <AppContent text="create survey" />
    </Route>
  </React.Fragment>
);

export default AppRoutes;
