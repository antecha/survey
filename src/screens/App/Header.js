import React from 'react';
import { Layout, Icon } from 'antd';

const Header = ({ collapsed, onIconClick }) => (
  <Layout.Header style={{ background: '#fff', padding: 0 }}>
    <Icon
      className="trigger"
      type={collapsed ? 'menu-unfold' : 'menu-fold'}
      onClick={onIconClick}
    />
  </Layout.Header>
);

export default Header;
