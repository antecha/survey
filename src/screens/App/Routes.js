// @flow
import React from 'react';
import { Route } from 'react-router-dom';
import CreateSurvey from '../CreateSurvey';
import { StyledContent } from './styles';
import type { RoutesProps } from './types';
import type { Element } from './types';

export const AppContent = ({ children }: RoutesProps): Element<'StyledContent'> => (
  <StyledContent>{children}</StyledContent>
);

const Routes = (): Element<any> => {
  return (
    <React.Fragment>
      <Route exact path="/">
        <AppContent>Suvey-list</AppContent>
      </Route>

      <Route path="/create-survey">
        <AppContent>
          <CreateSurvey />
        </AppContent>
      </Route>
    </React.Fragment>
  );
};

export default Routes;
