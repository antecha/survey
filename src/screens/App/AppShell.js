import React from 'react';
import { Layout } from 'antd';
import styled from 'styled-components';

import AppMenu from './AppMenu';
import AppRoutes from './AppRoutes';
import AppHeader from './AppHeader';

const StyledLayout = styled(Layout)`
  height: 100vh;
  .trigger {
    font-size: 18px;
    line-height: 64px;
    padding: 0 24px;
    cursor: pointer;
    transition: color 0.3s;
    &:hover {
      color: #1890ff;
    }
  }

  .logo {
    height: 32px;
    background: rgba(255, 255, 255, 0.2);
    margin: 16px;
  }
`;
class AppShell extends React.Component {
  state = {
    collapsed: false,
  };

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    return (
      <StyledLayout>
        <AppMenu collapsed={this.state.collapsed} />

        <Layout>
          <AppHeader collapsed={this.state.collapsed} onIconClick={this.toggle} />

          <AppRoutes />
        </Layout>
      </StyledLayout>
    );
  }
}

export default AppShell;
