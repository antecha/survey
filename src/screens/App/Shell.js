// @flow
import React from 'react';
import { Layout } from 'antd';
import Sidebar from './Sidebar';
import Routes from './Routes';
import Header from './Header';
import { StyledLayout } from './styles';
import type { ShellState } from './types';
import type { Element } from './types';

class Shell extends React.Component<{}, ShellState> {
  state = {
    collapsed: false,
  };

  toggle = (): void => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render(): Element<'StyledLayout'> {
    return (
      <StyledLayout>
        <Sidebar collapsed={this.state.collapsed} />

        <Layout>
          <Header collapsed={this.state.collapsed} onIconClick={this.toggle} />
          <div>tttettatadsadsadsasa</div>
          <Routes />
        </Layout>
      </StyledLayout>
    );
  }
}

export default Shell;
