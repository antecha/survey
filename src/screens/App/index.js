import React from "react";
import AppShell from './AppShell'

const App = () => <div><AppShell/></div>;

export default App;
