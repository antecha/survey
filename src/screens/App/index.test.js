import React from 'react';
import { create } from 'react-test-renderer';
import App from './index';
import { MemoryRouter } from 'react-router';

jest.mock('antd', _ => ({
  ...require.requireActual('antd'),
  Tabs: () => <div />,
}));

describe('App', () => {
  const testRenderer = create(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
