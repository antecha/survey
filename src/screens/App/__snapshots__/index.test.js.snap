// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<section
  className="sc-AxjAm jCuiTa ant-layout ant-layout-has-sider"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      />
      <ul
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-item ant-menu-item-selected"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <i
            aria-label="icon: user"
            className="anticon anticon-user"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </i>
          <span>
            Survey List
          </span>
          <a
            href="/"
            onClick={[Function]}
          />
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          skey="2"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <i
            aria-label="icon: video-camera"
            className="anticon anticon-video-camera"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="video-camera"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226zM208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </i>
          <span>
            create-survey
          </span>
          <a
            href="/create-survey"
            onClick={[Function]}
          />
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="ant-layout"
  >
    <header
      className="sc-AxirZ cnDfgi ant-layout-header"
    >
      <i
        aria-label="icon: menu-fold"
        className="anticon anticon-menu-fold trigger"
        onClick={[Function]}
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="menu-fold"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
          />
        </svg>
      </i>
    </header>
    <main
      className="sc-AxiKw jAJPxZ ant-layout-content"
    >
      Suvey-list
    </main>
  </section>
</section>
`;
